velox/exec/RowContainer.h

```diff
@@ -993,56 +1184,130 @@ inline void RowContainer::storeNoNulls<TypeKind::MAP>(
     FlatVector<StringView>* FOLLY_NONNULL result) {
-  result->resize(numRows);
-  for (int32_t i = 0; i < numRows; ++i) {
-    if (rows[i] == nullptr) {
-      result->setNull(i, true);
-    } else {
-      result->setNull(i, false);
-      extractString(valueAt<StringView>(rows[i], offset), result, i);
-    }
-  }
+  extractValuesNoNullsStringView<true>(
+      rows, rowNumbers, numRows, columnOffset, resultOffset, result);
 }
 
 template <>
-inline void RowContainer::extractValuesWithNulls<StringView>(
+inline void RowContainer::extractValuesNoNulls<false, StringView>(
     const char* FOLLY_NONNULL const* FOLLY_NONNULL rows,
+    folly::Range<const vector_size_t*> rowNumbers,
     int32_t numRows,
-    int32_t offset,
+    int32_t columnOffset,
+    int32_t resultOffset,
+    FlatVector<StringView>* FOLLY_NONNULL result) {
+  extractValuesNoNullsStringView<false>(
+      rows, rowNumbers, numRows, columnOffset, resultOffset, result);
+}
+
+template <>
+inline void RowContainer::extractValuesWithNulls<false, StringView>(
+    const char* FOLLY_NONNULL const* FOLLY_NONNULL rows,
+    folly::Range<const vector_size_t*> rowNumbers,
+    int32_t numRows,
+    int32_t columnOffset,
     int32_t nullByte,
     uint8_t nullMask,
+    int32_t resultOffset,
     FlatVector<StringView>* FOLLY_NONNULL result) {
-  result->resize(numRows);
-  for (int32_t i = 0; i < numRows; ++i) {
-    if (!rows[i] || isNullAt(rows[i], nullByte, nullMask)) {
-      result->setNull(i, true);
-    } else {
-      extractString(valueAt<StringView>(rows[i], offset), result, i);
-    }
-  }
+  extractValuesWithNullsStringView<false>(
+      rows,
+      rowNumbers,
+      numRows,
+      columnOffset,
+      nullByte,
+      nullMask,
+      resultOffset,
+      result);
+}
+
+template <>
+inline void RowContainer::extractValuesWithNulls<true, StringView>(
+    const char* FOLLY_NONNULL const* FOLLY_NONNULL rows,
+    folly::Range<const vector_size_t*> rowNumbers,
+    int32_t numRows,
+    int32_t columnOffset,
+    int32_t nullByte,
+    uint8_t nullMask,
+    int32_t resultOffset,
+    FlatVector<StringView>* FOLLY_NONNULL result) {
+  extractValuesWithNullsStringView<true>(
+      rows,
+      rowNumbers,
+      numRows,
+      columnOffset,
+      nullByte,
+      nullMask,
+      resultOffset,
+      result);
 }
 
 template <>
 inline void RowContainer::extractColumnTyped<TypeKind::OPAQUE>(
     const char* FOLLY_NONNULL const* FOLLY_NONNULL /*rows*/,
+    folly::Range<const vector_size_t*> /*rowNumbers*/,
     int32_t /*numRows*/,
     RowColumn /*column*/,
-    VectorPtr /*result*/) {
+    int32_t /*resultOffset*/,
+    VectorPtr& /*result*/) {
   VELOX_UNSUPPORTED("RowContainer doesn't support values of type OPAQUE");
 }
 
 inline void RowContainer::extractColumn(
     const char* FOLLY_NONNULL const* FOLLY_NONNULL rows,
     int32_t numRows,
     RowColumn column,
-    VectorPtr result) {
+    VectorPtr& result) {
+  VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
```

Comment:
Can we simplify this by calling `extractColumn(rows, numRows, column 0, result);` ?

Reply:
I've removed this extra signature.